### DISCUSSION TITLE
Validate form before submit

### DISCUSSION THREAD

**Discussion by checkifexist** [ansibleguy76/ansibleforms#162] [Q&A] [answered]
Can we validate form before triggering Run button, Not validating individual field but complete form as one,
for eg - I have 2 field, I want to make sure user has filled either first or second field?

**Comment by ansibleguy76** [comment, marked as answer]
Hey, it will be a part of 5.0.1.   The plan is to give live feedback on validity and issues.  It will mainly be an icon, not a second button.
I basically validates all the time anyhow.

**Comment by ansibleguy76** [comment]
Additional input on this :
5.0.1 now shows info when the form has issue (unresolved expressions for example).  The form is non-stop evaluated (every 100ms and on changes briefly every 4ms).  however I must do a final validation when the "run" button is clicked as it is possible that at that exact point you were still on a field you just changed that triggers a series of dependency-changes.  So it's not possible to disable the run button.  hence from time to time you will see the message "stabilizing form".
